### PR TITLE
view park

### PR DESCRIPTION
**Pull request by NguyennKhoaa**
view park cua em

### CODE REVIEW

File: lib/configs/routes/routes.dart [hunk 1]
```diff
@@ -1,10 +1,26 @@
...
+  static const String landingPage = "LandingPage";
+
+  static const String homePage = "HomePage";
+
+  static const String signInPage = "SignInPage";
+
+  static const String signUpPage = "SignUpPage";
+
+  static const String viewPark = "ViewPark";
+}
```
Comment (DQLok) [inline lib/configs/routes/routes.dart]: rebase this file OK!!!!

File: lib/main.dart [hunk 2]
```diff
@@ -19,9 +22,12 @@ class MyApp extends StatelessWidget {
...
-          //home: const RootPage()
+           initialRoute: "/ViewPark",
+        routes: {
+          "/": (context) => const RootPage(),
+          "/userProfile": (context) => const UserProfile(),
+          "/ViewPark": (context) => const ViewPark()
+        },
           ),
     );
   }
```
Comment (DQLok) [inline lib/main.dart:33]: rebase this file OK!!!!

File: lib/view/viewPark/view_park.dart [hunk 3]
```diff
@@ -0,0 +1,85 @@
...
+  _ViewParkState createState() => _ViewParkState();
+}
+
+class _ViewParkState extends State<ViewPark> {
+  @override
+  Widget build(BuildContext context) {
+    return Scaffold(
+      appBar: AppBar(
+        backgroundColor: AppColor.blueText,
+        iconTheme: const IconThemeData(color: Color(0xff3277DB)),
```
Comment (DQLok) [inline lib/view/viewPark/view_park.dart]: you should create color in AppColor and use it

File: lib/view/viewPark/view_park.dart [hunk 4]
```diff
@@ -0,0 +1,85 @@
...
+        elevation: 0,
+      ),
+      body: Column(
+        children: [Container(
+            constraints: const BoxConstraints.expand(height: 280),
```
Comment (DQLok) [inline lib/view/viewPark/view_park.dart]: use => Size size = MediaQuery.of(context).size;  for Height and Width

File: lib/view/viewPark/view_park.dart [hunk 5]
```diff
@@ -0,0 +1,85 @@
...
+            color: AppColor.blueText,
+            child: Column(
+              mainAxisAlignment: MainAxisAlignment.center,
+              children: <Widget>[
+                const SizedBox(
+                  height: 40,
```
Comment (DQLok) [inline lib/view/viewPark/view_park.dart]: use => Size size = MediaQuery.of(context).size;  for Height and Width

File: lib/view/viewPark/view_park.dart [hunk 6]
```diff
@@ -0,0 +1,85 @@
...
+                ),
+                Image.asset("assets/images/xe.png"),
```
Comment (DQLok) [inline lib/view/viewPark/view_park.dart]: all widget use SizeBox(child: /*your widget*/)

File: lib/view/viewPark/view_park.dart [hunk 7]
```diff
@@ -0,0 +1,85 @@
...
+                Padding(
+                  padding: const EdgeInsets.fromLTRB(0, 40, 0, 40),
+                  child: RichText(
+                    text: const TextSpan(
+                    text: "Parking App - ",
+                    style: TextStyle(color: Color(0xff606470), fontSize: 25, fontWeight: FontWeight.bold),
+                    children: <TextSpan>[
+                      TextSpan(
+                        text: "20/hr",
+                        style: TextStyle(color: Color(0xff606470), fontSize: 16))
```
Comment (DQLok) [inline lib/view/viewPark/view_park.dart]: you should create color in AppColor and use it

File: lib/view/viewPark/view_park.dart [hunk 8]
```diff
@@ -0,0 +1,85 @@
...
+                const SizedBox(
+                  height: 40,
+                ),
+                Image.asset("assets/images/xe.png"),
+                Padding(
+                  padding: const EdgeInsets.fromLTRB(0, 40, 0, 40),
+                  child: RichText(
+                    text: const TextSpan(
+                    text: "Parking App - ",
+                    style: TextStyle(color: Color(0xff606470), fontSize: 25, fontWeight: FontWeight.bold),
```
Comment (DQLok) [inline lib/view/viewPark/view_park.dart]: you should create color in AppColor and use it

File: lib/view/viewPark/view_park.dart [hunk 9]
```diff
@@ -0,0 +1,85 @@
...
+                    ),
+                ),
+              ],
+            ),
+          ),
+          const Padding(
+                padding: EdgeInsets.fromLTRB(0, 15, 0, 0),
+                child: Text(
+                  "End time",
+                  style: TextStyle(fontSize: 22, fontWeight: FontWeight.bold, color: Color(0xff333333)),
```
Comment (DQLok) [inline lib/view/viewPark/view_park.dart]: you should create color in AppColor and use AppTextStyle

File: lib/view/viewPark/view_park.dart (hunk 7 above)
Comment (DQLok) [reply lib/view/viewPark/view_park.dart]: use AppTextStyle

File: lib/view/viewPark/view_park.dart [hunk 10]
```diff
@@ -0,0 +1,85 @@
...
+                ),
+                ),
+                    Container(
+                        margin: const EdgeInsets.only(top: 15),
+                        child: Row(
+                          children: [
+                            const SizedBox(width: 10,),
```
Comment (DQLok) [inline lib/view/viewPark/view_park.dart]: use => Size size = MediaQuery.of(context).size; for Height and Width

File: lib/view/viewPark/view_park.dart [hunk 11]
```diff
@@ -0,0 +1,85 @@
...
+                            SizedBox(
+                              // margin: const EdgeInsets.only(right: 20),
+                              child: ButtonViewPark(content: "Today", voidCallBack: (){})
+                            ),
+                            const SizedBox(width: 45,),
```
Comment (DQLok) [inline lib/view/viewPark/view_park.dart]: use => Size size = MediaQuery.of(context).size; for Height and Width

File: lib/view/viewPark/view_park.dart [hunk 12]
```diff
@@ -0,0 +1,85 @@
...
+                            SizedBox(
+                              // margin: const EdgeInsets.only(left: 15),
+                              child: ButtonViewPark(content: "Tomorrow", voidCallBack: (){})
+                            ),
+                            const SizedBox(width: 45,),
```
Comment (DQLok) [inline lib/view/viewPark/view_park.dart]: use => Size size = MediaQuery.of(context).size; for Height and Width

File: lib/view/viewPark/view_park.dart [hunk 13]
```diff
@@ -0,0 +1,85 @@
...
+                            SizedBox(
+                              // margin: const EdgeInsets.only(left: 70),
+                              child: ButtonViewPark(content: "Later", voidCallBack: (){})
+                            )
+                          ],
+                        ),
+                      ),
+                      const SizedBox(height: 130),
```
Comment (DQLok) [inline lib/view/viewPark/view_park.dart]: use => Size size = MediaQuery.of(context).size; for Height and Width

File: lib/view/viewPark/view_park.dart [hunk 14]
```diff
@@ -0,0 +1,85 @@
+import 'package:flutter/material.dart';
+import 'package:parkingappmobile/configs/themes/app_color.dart';
+import 'package:parkingappmobile/widgets/button/button_view_park.dart';
+import 'package:slide_to_confirm/slide_to_confirm.dart';
+
+class ViewPark extends StatefulWidget {
```
Comment (baosphoa0403) [inline lib/view/viewPark/view_park.dart]: Can you explain for me why Stateful ?

File: lib/widgets/button/button_view_park.dart [hunk 15]
```diff
@@ -0,0 +1,33 @@
+import 'package:flutter/material.dart';
+import 'package:parkingappmobile/configs/themes/app_color.dart';
+import 'package:parkingappmobile/configs/themes/app_text_style.dart';
+
+class ButtonViewPark extends StatelessWidget {
```
Comment (baosphoa0403) [inline lib/widgets/button/button_view_park.dart]: Please explain why don't use ButtonBase

**Review by baosphoa0403 — CHANGES_REQUESTED** [review]
Please tell me about it

**Review by DQLok — APPROVED** [review]

**Review by baosphoa0403 — APPROVED** [review]
review done

**Pull request merged**